livekit/src/rtc_engine/rtc_session.rs

```diff
@@ -778,8 +798,8 @@ impl SessionInner {
     async fn simulate_scenario(&self, scenario: SimulateScenario) -> EngineResult<()> {
         let simulate_leave = || {
             self.on_signal_event(proto::signal_response::Message::Leave(proto::LeaveRequest {
+                action: proto::leave_request::Action::Reconnect.into(),
```

Comment:
it doesn't look like this is handled? this would determine if a client should reconnect vs disconnect for good.

Reply:
from what i've seen rust-sdks reconnection logic is fundamentally different, and doesn't use `LeaveRequest` for retrying at all? it attempts to restart the publisher, or completely closing it (with `Disconnect`) then reconnecting and recovering? i'll see if i can make it consistent with JS

Reply:
I think the key here is if server is not sending a reconnect action, Rust should not attempt to reconnect. CC: @theomonnom

Reply:
also can confirm streams are received and handled correctly